weblate/api/views.py

```diff
@@ -1212,6 +1212,45 @@ def get_serializer(self, instance, *args, **kwargs):
     def get_queryset(self):
         return Unit.objects.filter_access(self.request.user).order_by("id")
 
+    @action(detail=True, methods=["post"])
+    def screenshots(self, request, **kwargs):
+        obj = self.get_object()
+
+        if not request.user.has_perm("unit.edit", obj.translation):
```

Comment:
It should check the screenshot permission:

```suggestion
        if not request.user.has_perm("screenshot.edit", obj.translation):
```